lib/utils/helpers.ts

```diff
@@ -187,35 +188,6 @@ export function getApplication(applicationName: string) {
 	return balena.models.application.get(applicationName, extraOptions);
 }
 
-/**
- * Choose between 'cmd.exe' and '/bin/sh' for running the given command string,
- * depending on the value of `os.platform()`.
- * When writing new code, consider whether it would be possible to avoid using a
- * shell at all, using the which() function in this module to obtain a program's
- * full path, executing the program directly and passing the arguments as an
- * array instead of a long string. Avoiding a shell has several benefits:
- *   - Avoids the need to shell-escape arguments, especially nested commands.
- *   - Bypasses the incompatibilities between cmd.exe and /bin/sh.
- *   - Reduces the security risks of lax input validation.
- * Code example avoiding a shell:
- *    const program = await which('ssh');
- *    const args = ['root@192.168.1.1', 'cat /etc/os-release'];
- *    const child = spawn(program, args);
- */
-export function getSubShellCommand(command: string) {
```

Comment:
This function is being deleted because, of course, there was not any remaining usage of it, but also because I think it encourages a bad pattern of executing a shell in the first place. In the past I had already fixed bugs because of the usage of this function, and now again in this PR. All previous invocations of shells were limited to taking advantage of a shell's ability to select the full path of an executable given its "short name", like 'ssh' instead of `C:\WINDOWS\System32\OpenSSH\ssh.EXE`, at the cost of us having to deal with argument quoting and escaping in a shell-specific way. The `which()` function + spawn with argument in an array (instead of a plain string) is a better pattern.